form/src/main/java/org/dhis2/form/ui/FormViewModel.kt

```diff
@@ -133,6 +137,12 @@ class FormViewModel(
                         Timber.d("${result.first.id} is changing its value")
                         _queryData.value = it
                     }
+                    if (repository.hasLegendSet(result.first.id)) {
```

Comment:
is this `hasLegendSet` a heavy request , might it block the main thread?.
If so, maybe we can create a used case with a coroutine to check and perform the action

Reply:
This is running on the IO thread :)